### PR TITLE
Handle error and loading states from productSearch query

### PR DESCRIPTION
**Pull request by victorhmp**
#### What is the purpose of this pull request?

Add error handling logic to `ProductSummaryList` for the `productSearch` query.

#### What problem is this solving?

`ProductSummaryList` could break while loading the query.

#### How should this be manually tested?

https://shelflayout--storecomponents.myvtex.com/

#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/ProductSummaryList.js [hunk 1]
```diff
@@ -80,6 +80,11 @@ const ProductSummaryList = ({
     },
   })
 
+  // TODO Improve this error handling in the future
```
Comment (Klynger) [inline react/ProductSummaryList.js]: This could be an actual issue and have the link right here

File: store/interfaces.json [hunk 2]
```diff
@@ -86,6 +86,10 @@
   "list-context.product-list": {
     "allowed": ["product-summary"],
     "composition": "children",
-    "component": "ProductSummaryList"
+    "component": "ProductSummaryList",
+    "preview": {
```
Comment (Klynger) [inline store/interfaces.json]: Is this related to the PR? 🤔

Comment (victorhmp) [reply store/interfaces.json]: Yep

File: react/ProductSummaryList.js [hunk 3]
```diff
@@ -80,6 +80,11 @@ const ProductSummaryList = ({
     },
   })
 
+  // https://github.com/vtex-apps/product-summary/issues/235
```
Comment (Klynger) [inline react/ProductSummaryList.js]: ✨
